src/main/java/org/opentripplanner/middleware/auth/Auth0Users.java

```diff
@@ -0,0 +1,355 @@
+        URI uri;
+        try {
+            uri = builder.build();
+        } catch (URISyntaxException e) {
+            LOG.error("Unable to build URI to getUserById");
+            e.printStackTrace();
+            return null;
+        }
+        String response = doRequest(uri);
+        if (response == null) {
+            LOG.error("Auth0 request aborted due to issues during request.");
+            return null;
+        }
+        Auth0UserProfile user = null;
+        try {
+            user = mapper.readValue(response, Auth0UserProfile.class);
+        } catch (IOException e) {
+            LOG.error("Unable to parse user profile response from Auth0! Response: {}", response);
+            e.printStackTrace();
+        }
+        return user;
+    }
+
+    /**
+     * Get a single Auth0 user for the specified email.
+     */
+    public static Auth0UserProfile getUserByEmail(String email) {
+        URIBuilder builder = getURIBuilder();
+        builder.setPath(USERS_API_PATH + "-by-email");
+        builder.setParameter("search_engine", SEARCH_API_VERSION);
+        builder.setParameter("email", email);
+        URI uri;
+        try {
+            uri = builder.build();
+        } catch (URISyntaxException e) {
+            LOG.error("Unable to build URI to getUserById");
+            e.printStackTrace();
+            return null;
+        }
+        String response = doRequest(uri);
+        if (response == null) {
+            LOG.error("Auth0 request aborted due to issues during request.");
+            return null;
+        }
+        try {
+            Auth0UserProfile[] userProfiles = mapper.readValue(response, Auth0UserProfile[].class);
+            if (userProfiles.length > 0) {
+                return userProfiles[0];
+            }
+        } catch (IOException e) {
+            LOG.error("Unable to parse user profile response from Auth0! Response: {}", response);
+            e.printStackTrace();
+        }
+        return null;
+    }
+
+    /**
+     * Creates a new uri builder and sets the scheme, port and host according to whether a test environment is in effect
+     */
+    private static URIBuilder getURIBuilder() {
+        URIBuilder builder = new URIBuilder();
+        if (AUTH0_DOMAIN.equals("your-auth0-domain")) {
+            // set items for testing purposes assuming use of a Wiremock server
+            builder.setScheme("http");
+            builder.setPort(8089);
+            builder.setHost("localhost");
+        } else {
+            // use live Auth0 domain
+            builder.setScheme("https");
+            builder.setHost(AUTH0_DOMAIN);
+        }
+        return builder;
+    }
+
+    /**
+     * Get users subscribed to a given target ID.
+     */
+    public static String getUsersBySubscription(String subscriptionType, String target) {
+        return getAuth0Users("app_metadata.datatools.subscriptions.type:" + subscriptionType + " AND app_metadata.datatools.subscriptions.target:" + target);
```

Comment:
Will these subscriptions be a thing in FDOT? If not, let's remove. We should also move away from storing user data in the auth0 app_metadata as well.

Reply:
Notifications will definitely be a thing. But we won't be storing things in app_metadata. I think we should just plan to use the user collection for this stuff. We need to figure out how we want to store various user types (users of OTP vs. users of admin dashboard).